azurelinuxagent/common/utils/cryptutil.py

```diff
@@ -128,3 +141,17 @@ def bits_to_bytes(self, bits):
                 index = 7
         return bytes(byte_array)
 
+    def decodeBase64ToFile(self, cacheFile, base64str):
+        with open(cacheFile, "wb") as c:
+            c.write(base64.b64decode(base64str))
```

Comment:
There's a b64decode method in textutil, which provides better protection for Python2 vs. Python3.

Reply:
removed and changed to use that.